src/types/errors/custom-errors.ts

```diff
@@ -0,0 +1,28 @@
+/*
```

Comment:
Better to include one error per file, and name the file based on the error `HttpNotSupportedError.ts`. Also thinking more about it, these are not types, but classes, so we could move them into `src/errors` directly instead of keeping them under types.